src/ml/federated/JasmineGraphFederatedInstance.cpp

```diff
@@ -0,0 +1,483 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <map>
+#include <iostream>
+#include "JasmineGraphFederatedInstance.h"
+#include "../../server/JasmineGraphInstanceProtocol.h"
+#include "../../util/Utils.h"
+#include "../trainer/JasmineGraphTrainingSchedular.h"
+#include "../../server/JasmineGraphServer.h"
+#include "../../util/logger/Logger.h"
+#include "../../util/Utils.h"
+
+
+Logger fed_trainer_log;
+
+void JasmineGraphFederatedInstance::initiateFiles(std::string graphID, std::string trainingArgs) {
+
+    int count = 0;
+    JasmineGraphTrainingSchedular *schedular = new JasmineGraphTrainingSchedular();
+    map<string, map<int, int>> scheduleForAllHosts = schedular->schedulePartitionTraining(graphID);
+    JasmineGraphServer *jasmineServer = new JasmineGraphServer();
+    std::map<std::string, JasmineGraphServer::workerPartitions> graphPartitionedHosts = jasmineServer->getGraphPartitionedHosts(
+            graphID);
+    int partition_count = 0;
+    std::map<std::string, JasmineGraphServer::workerPartitions>::iterator mapIterator;
+    for (mapIterator = graphPartitionedHosts.begin(); mapIterator != graphPartitionedHosts.end(); mapIterator++) {
+        JasmineGraphServer::workerPartitions workerPartition = mapIterator->second;
+        std::vector<std::string> partitions = workerPartition.partitionID;
+        std::vector<std::string>::iterator it;
+        for (it = partitions.begin(); it < partitions.end(); it++) {
+            partition_count++;
+        }
+    }
+    std::cout << partition_count << std::endl;
+    std::thread *workerThreads = new std::thread[partition_count];
+
+    Utils utils;
+    string prefix = utils.getJasmineGraphProperty("org.jasminegraph.server.instance.trainedmodelfolder");
+    string attr_prefix = utils.getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+    trainingArgs = trainingArgs;
+    std::map<std::string, JasmineGraphServer::workerPartitions>::iterator j;
+    for (j = graphPartitionedHosts.begin(); j != graphPartitionedHosts.end(); j++) {
+        JasmineGraphServer::workerPartitions workerPartition = j->second;
+        std::vector<std::string> partitions = workerPartition.partitionID;
+        string partitionCount = std::to_string(partitions.size());
+        std::vector<std::string>::iterator k;
+        map<int, int> scheduleOfHost = scheduleForAllHosts[j->first];
+        for (k = partitions.begin(); k != partitions.end(); k++) {
+            int iterationOfPart = scheduleOfHost[stoi(*k)];
+            workerThreads[count] = std::thread(initiateTrain, j->first, workerPartition.port, workerPartition.dataPort,trainingArgs + " " + *k,iterationOfPart, partitionCount);
```

Comment:
Break this line into two. The line is too lengthy.

Reply:
Fixed in ffb1fdb

Reply:
Done